packages/blade/src/components/Skeleton/_decisions/decisions.md

```diff
@@ -0,0 +1,320 @@
+# Skeleton <!-- omit in toc -->
+Skeleton Loader is a static / animated placeholder for the information that is still loading. It mimic the structure and look of the entire view. 
+
+<img src="./skeleton-thumbnail.png" width="380" />
+
+## Design
+- [Figma - Skeleton](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=16508%3A258522&mode=dev)
+
+
+## API
+
+| Prop     | Type        | Default | Description                                  | Required |
+| -------- | ----------- | ------- | -------------------------------------------- | -------- |
+| contrast | `low, high` | `low`   | The contrast of the Skeleton to be rendered. |          |
+
+Skeleton will have subset of Box props to help users layout the Skeleton blocks accordingly. 
+
+Layout: 
+
+- borderRadius
+- height
+- minHeight
+- maxHeight
+- width
+- minWidth
+- maxWidth
+
+Flex:
+
+- flexGrow
+- flexShrink
+- flexBasis
+- justifyContent
+- justifySelf
+- placeSelf
+- order
+
+Margin: 
+
+- margin
+- margin {top | bottom | left | right | x | y}
+
+Position:
+
+- position
+- top
+- right
+- bottom
+- left
+
+
+### Sample Usage
+
+Basic usage: 
+
+```jsx
+<Skeleton width="100%" height="20px" borderRadius="medium" marginY="spacing.1" />
+<Skeleton width="100%" height="20px" borderRadius="medium" marginY="spacing.1" />
+<Skeleton width="100%" height="20px" borderRadius="medium" marginY="spacing.1" />
+```
+
+<img src="./skeleton-usage-demo-1.png" alt="Skeleton basic usage" width="50%" />
+
+Complex usage: 
+
+```jsx
+<Box
+  display="flex"
+  flexDirection="column"
+  gap="spacing.2"
+>
+  <Box display="flex" flexDirection="row" gap="spacing.2">
+    <Skeleton
+      flexShrink={0}
+      width="40px"
+      height="40px"
+      borderRadius="max"
+    />
+    <Skeleton width="100%" />
+  </Box>
+  <Box flexDirection="column" display="flex" gap="spacing.2">
+    <Skeleton />
+    <Skeleton />
+  </Box>
+</Box>
+```
+
+<img src="./skeleton-usage-demo-2.png" alt="Skeleton complex usage" width="50%" />
+
+
+### Working with existing components. 
+
+To use Skeleton loader with existing blade components like Card, consumers can create custom skeleton templates as per their needs and usecases. 
+
+**Example:** 
+
+If the whole card needs to be loadable: 
+
+```jsx
+const App = () => {
+  const { data, isLoading } = useQuery();
+
+  return (
+    <>
+      {isLoading ? (
+        <Box>
+          <Box display="flex">
+            <Box>
+              <Skeleton width="40%" height="30px" />
+              <Skeleton width="30%" height="20px" />
+            </Box>
+            <Skeleton width="30px" height="50px" />
+          </Box>
+          <Skeleton width="100%" height="100px" />
+        </Box>
```

Comment:
Sorry forgot the elevation prop. We've added that on Box for this exact usecase. 

elevation="lowRaised" borderRadius="medium" will suffice.

Reply:
There will be certain other things such as padding in card so users will have to match those in this `Box` wrapper. If card padding changes in a future release users will have to manually update such Box occurences (which is why I was suggesting this example might not be ideal - it recommends a practice to replicate a card by creating their own box wrapper.

Can we replace the outer `Box` by a `Card`?

Reply:
> it recommends a practice to replicate a card by creating their own box wrapper.

I see, can update the examples. but agree that in general people will need to make reusable composite components for skeleton loaders, otherwise lot of copy pasting will happening.

Reply:
> Can we replace the outer Box by a Card?

We have both examples - 1. Skeleton inside `<Card />`  2. Skeleton inside `<CardBody />`

Since the root `<Card />` doesn't allow any other components as it's children other than Card specific ones, if consumer wants to make the whole card as a skeleton they will need to recreate the wrapper also.

Reply:
Need for recreating the wrapper seems non-ideal in terms of DX (and ensuring consistency eg. padding, etc.) 🤔